client/src/components/UploadModal/ImageUploader.js

```diff
@@ -0,0 +1,134 @@
+import PropTypes from 'prop-types';
+import { makeStyles } from '@material-ui/core/styles';
+import Avatar from '@material-ui/core/Avatar';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Dialog from '@material-ui/core/Dialog';
+import { orange } from '@material-ui/core/colors';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+import Alert from '@material-ui/lab/Alert';
+
+import axios from 'axios';
+import FormData from 'form-data';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    margin: theme.spacing(1),
+  },
+  avatar: {
+    backgroundColor: orange[100],
+    color: orange[600],
+    width: theme.spacing(22),
+    height: theme.spacing(22),
+    margin: '2rem 5rem',
+  },
+  input: {
+    display: 'none',
+  },
+  button: {
+    marginBottom: theme.spacing(3) 
+  }
+}));
+
+export default function UploadDialog(props) {
+  const { onClose, open } = props;
+  const classes = useStyles();
+
+  const handleClose = () => {
+    onClose();
+  };
+
+  const [file, setFile] = useState('');
+  const [filename, setFilename] = useState('Choose File');
+  const [uploadedFile, setUploadedFile] = useState({});
+  const [message, setMessage] = useState('');
+
+  const onChange = (e) => {
+    setFile(e.target.files[0]);
+    setFilename(e.target.files[0].name);
+  };
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    const formData = new FormData();
+    formData.append('file', file);
+
+    try {
+      const res = await axios.post('/api/image-upload', formData, {
+        headers: {
+          'Content-Type': 'multipart/form-data',
+        },
+      });
+
+      const { fileName, filePath } = res.data;
+
+      setUploadedFile({ fileName, filePath });
+      console.log('File Uploaded');
+      setMessage(true)
+      await setTimeout(()=>{
+        setMessage(false)
+        handleClose();
+        }, 2000);
+    } catch (err) {
+      if (err.response.status === 500) {
+        console.log('There was a problem with the server');
+      } else {
+        console.log(err.response.data.msg);
+      }
+    }
+  };
+
+  return (
+    <Dialog
+      className={classes.root}
+      onClose={handleClose}
+      aria-labelledby='simple-dialog-title'
+      open={open}
+    >
+      <DialogTitle id='simple-dialog-title'>Update profile photo</DialogTitle>
+
+      {message && <Alert>Your profile photo was updated!</Alert>}
+
+      <Avatar className={classes.avatar} />
+
+      <form onSubmit={onSubmit}>
+        <Box display="flex" justifyContent="space-around">
+        <span className={classes.button}>
+          <input
+            type='file'
+            accept='image/*'
+            className={classes.input}
+            id='upload-button-file'
+            multiple
+            onChange={onChange}
+          />
+          <label htmlFor='upload-button-file'>
+            <Button variant='contained' color='secondary' component='span'>
+              + Upload
```

Comment:
let's use icons instead of text for this type of thing
[docs](https://material-ui.com/components/buttons/#buttons-with-icons-and-label)